lib/modules/datasource/nuget/v2.ts

```diff
@@ -6,65 +6,71 @@ import { regEx } from '../../../util/regex';
 import type { ReleaseResult } from '../types';
 import { massageUrl, removeBuildMeta } from './common';
 
-function getPkgProp(pkgInfo: XmlElement, propName: string): string | undefined {
-  return pkgInfo.childNamed('m:properties')?.childNamed(`d:${propName}`)?.val;
-}
+export class NugetV2Api {
+  getPkgProp(pkgInfo: XmlElement, propName: string): string | undefined {
+    return pkgInfo.childNamed('m:properties')?.childNamed(`d:${propName}`)?.val;
+  }
 
-export async function getReleases(
-  http: Http,
-  feedUrl: string,
-  pkgName: string,
-): Promise<ReleaseResult | null> {
-  const dep: ReleaseResult = {
-    releases: [],
-  };
-  let pkgUrlList: string | null = `${feedUrl.replace(
-    regEx(/\/+$/),
-    '',
-  )}/FindPackagesById()?id=%27${pkgName}%27&$select=Version,IsLatestVersion,ProjectUrl,Published`;
-  while (pkgUrlList !== null) {
-    // typescript issue
-    const pkgVersionsListRaw: HttpResponse<string> = await http.get(pkgUrlList);
-    const pkgVersionsListDoc = new XmlDocument(pkgVersionsListRaw.body);
+  async getReleases(
+    http: Http,
+    feedUrl: string,
+    pkgName: string,
+  ): Promise<ReleaseResult | null> {
+    const dep: ReleaseResult = {
+      releases: [],
+    };
+    let pkgUrlList: string | null = `${feedUrl.replace(
+      regEx(/\/+$/),
+      '',
+    )}/FindPackagesById()?id=%27${pkgName}%27&$select=Version,IsLatestVersion,ProjectUrl,Published`;
+    while (pkgUrlList !== null) {
+      // typescript issue
+      const pkgVersionsListRaw: HttpResponse<string> =
```

Comment:
do we still need the explicit type here?

Reply:
Sharp eyes! Nope, that is not required. Removed in b152c05d02e365723cd83719928cc99971bc7fd2